tests/integration_tests/tests/agent_tests/test_snapshot_create.py

```diff
@@ -0,0 +1,135 @@
+########
+# Copyright (c) 2013-2019 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+import uuid
+
+from integration_tests import AgentTestCase
+from cloudify.models_states import AgentState
+from integration_tests.tests.utils import (get_resource as resource,
+                                           create_rest_client)
+
+CREATE_SNAPSHOT_SUCCESS_MSG =\
+    "'create_snapshot' workflow execution succeeded"
+RESTORE_SNAPSHOT_SUCCESS_MSG =\
+    "'restore_snapshot' workflow execution succeeded"
+
+
+class TestSnapshotCreate(AgentTestCase):
+
+    def _deploy_with_agents(self, states, deployments, prefix):
+        for i in range(len(states)):
+            deployments.append(
+                self.deploy_application(resource(
+                    "dsl/agent_tests/with_agent.yaml"),
+                    deployment_id='{0}{1}'.format(prefix, i))[0]
+            )
+        for i in range(len(states)):
+            agent_id = self.client.agents.list(
+                deployment_id='{0}{1}'.format(prefix, i)
+            ).items[0].id
+            self.client.agents.update(agent_id, states[i])
+        self.assertEqual(
+            len(self.client.agents.list(state=states).items),
+            len(states)
+        )
+
+    def _create_snapshot(self):
+        snapshot_id = 's{0}'.format(uuid.uuid4())
+        execution = self.client.snapshots.create(snapshot_id, False, False)
+        self.wait_for_event(execution, CREATE_SNAPSHOT_SUCCESS_MSG)
+        return snapshot_id
+
+    def _restore_snapshot_and_validate(self, states, deployments, snapshot_id):
+        execution = self.client.snapshots.restore(snapshot_id, force=True)
+        # give the database some time to downgrade/upgrade before running
+        # requests to avoid the deadlock described in CY-1455
+        time.sleep(15)
```

Comment:
I think it should be more similar to the function `_wait_for_restore_execution_to_end` maybe we can move it to utils and use the same function.